ReactSkia/sdk/OnScreenKeyBoard.h

```diff
@@ -0,0 +1,175 @@
+typedef std::vector<std::vector<keyPosition_t>> KBLayoutKeyPosContainer;
+typedef std::vector<std::vector<keySiblingInfo_t>> KBLayoutSibblingInfoContainer;
+
+
+struct OSKLayout {
+    KBLayoutKeyInfoContainer*  keyInfo;
+    KBLayoutKeyPosContainer*    keyPos;
+    KBLayoutSibblingInfoContainer*    siblingInfo;
+    keyPlacementConfig_t*          KBGroupConfig;
+    KBLayoutType      KBLayoutType;
+    unsigned int      textFontSize;
+    unsigned int      textHLFontSize;
+    unsigned int      HorizontalStartOffset;
+    SkPoint           defaultFocussIndex;
+};
+
+class OnScreenKeyboard {
+
+    public:
+        static OnScreenKeyboard& getInstance(); // Interface to get OSK singleton object
+        static OSKErrorCode launch(OSKConfig oskConfig=defaultOSKConfig);// Interface to launch OSK
+        static void exit(); //Interface to quit OSK
+        static bool IsKBActive() {
+             OnScreenKeyboard &oskHandle=OnScreenKeyboard::getInstance();
+             return oskHandle.isOSKActive;
+        } // Interface to get whether is in launch or exit state
+
+    private:
+        OnScreenKeyboard() {
+            sem_init(&semPermitToLaunch,0,1);//Allow by default. Next signal to be done on exit call
+            sem_init(&semPermitToExit,0,0);//Block by default. signal to be done on launch Call
+        };
+        ~OnScreenKeyboard() {
+            sem_destroy(&semPermitToLaunch);
+            sem_destroy(&semPermitToExit);
+        };
+
+        void cleanUpOSKInstance();
+        void prepareToLaunch();
+        void createOSkWindow();
+        void showOSKWindow();
+        void pushToDisplay();
+
+        void onHWkeyHandler(rnsKey key, rnsKeyAction eventKeyAction);
+        void onExposeHandler(RnsShell::Window* window);
+
+        void createOSKLayout(OSKTypes KBtype );
+        void highlightFocussedKey(SkPoint index);
+        void handleSelection();
+        void drawOSK(OSKTypes oskType);
+        void drawPlaceHolder();
+        void drawFont(SkPoint index,SkColor color,bool onHLTile=false);
+        void drawOSKPartition();
+
+// Maintainables
+        RnsShell::PlatformDisplay::Type platformType;
+        RnsShell::Window* OSKwindow_;
+        std::unique_ptr<RnsShell::WindowContext> OSKwindowContext_;
+        sk_sp<SkSurface> backBuffer_;
+        SkCanvas *OSKcanvas_=nullptr;
+        int exposeEventID_{-1};
+        int OSKeventId_{-1};
+        sem_t semPermitToLaunch;
+        sem_t semPermitToExit;
+        sem_t semReadyToDraw;/* To sync expose event & window creation for x11 backend*/
+        bool  isOSKActive{false};
+
+        OSKConfig     oskConfig_;
+        OSKLayout     oskLayout_;
+        bool          generateOSKLayout{true};
+        SkSize        ScreenSize{0,0};
+        SkPoint       focussedKey_{0};
+        SkPoint       lastFocussedKey_{0};
+        SkColor       bgColor_{SK_ColorWHITE};
+        SkColor       fontColor_{SK_ColorWHITE};
```

Comment:
1. some variables starting with osk and some with OSK ... need to ensure uniformity
2. private variables to have '_' as prefix

Reply:
Will Change it

Reply:
On #2 , my bad .. its post fix with '_' .. few private variables are missing them here